Controller/Payment/Webhook.php

```diff
@@ -0,0 +1,446 @@
+     */
+    protected $logger;
+
+    /**
+     * @var \Magento\Framework\App\ObjectManager
+     */
+    protected $objectManagement;
+
+    /**
+     * @var \Magento\Quote\Model\QuoteManagement
+     */
+    protected $quoteManagement;
+
+    /**
+     * @var \Razorpay\Magento\Model\CheckoutFactory
+     */
+    protected $checkoutFactory;
+
+    /**
+     * @var \Magento\Quote\Model\QuoteRepository
+     */
+    protected $quoteRepository;
+
+    /**
+     * @var \Magento\Store\Model\StoreManagerInterface
+     */
+    protected $storeManagement;
+
+    /**
+     * @var \Magento\Customer\Api\CustomerRepositoryInterface
+     */
+    protected $customerRepository;
+
+    /**
+     * @var \Magento\Framework\Event\ManagerInterface
+     */
+    protected $eventManager;
+
+    /**
+     * @var \Magento\Sales\Model\Service\InvoiceService
+     */
+    protected $invoiceService;
+
+    /**
+     * @var \Magento\Framework\DB\Transaction
+     */
+    protected $transaction;
+
+    /**
+     * @var \Magento\Catalog\Model\Session
+     */
+    protected $catalogSession;
+
+    /**
+     * @var OrderRepositoryInterface
+     */
+    protected $orderRepository;
+
+    /**
+     * @var \Magento\Sales\Model\Order\Email\Sender\InvoiceSender
+     */
+    protected $invoiceSender;
+
+    /**
+     * @var \Magento\Sales\Model\Order\Email\Sender\OrderSender
+     */
+    protected $orderSender;
+
+    /**
+     * @var \Magento\Framework\App\CacheInterface
+     */
+    protected $cache;
+
+    /**
+     * @var STATUS_PROCESSING
+     */
+    protected const STATUS_PROCESSING = 'processing';
+
+    /**
+     * @param \Magento\Framework\App\Action\Context $context
+     * @param \Magento\Customer\Model\Session $customerSession
+     * @param \Magento\Checkout\Model\Session $checkoutSession
+     * @param \Razorpay\Magento\Model\Config $config
+     * @param \Magento\Sales\Api\Data\OrderInterface $order
+     * @param \Psr\Log\LoggerInterface $logger
+     * @param \Magento\Quote\Model\QuoteManagement $quoteManagement
+     * @param \Razorpay\Magento\Model\CheckoutFactory $checkoutFactory
+     * @param \Magento\Quote\Model\QuoteRepository $quoteRepository
+     * @param \Magento\Store\Model\StoreManagerInterface $storeManagement
+     * @param \Magento\Customer\Api\CustomerRepositoryInterface $customerRepository
+     * @param \Magento\Framework\Event\ManagerInterface $eventManager
+     * @param \Magento\Sales\Model\Service\InvoiceService $invoiceService
+     * @param \Magento\Framework\DB\Transaction $transaction
+     * @param \Magento\Catalog\Model\Session $catalogSession
+     * @param OrderRepositoryInterface $orderRepository
+     * @param \Magento\Sales\Model\Order\Email\Sender\InvoiceSender $invoiceSender
+     * @param \Magento\Sales\Model\Order\Email\Sender\OrderSender $orderSender
+     * @param \Magento\Framework\App\CacheInterface $cache
+     */
+    public function __construct(
+        \Magento\Framework\App\Action\Context $context,
```

Comment:
@enishant , please verify if that much dependency needs to be loaded.

Reply:
@ramth05 Removed unused dependencies